packages/transition-backend/src/models/db/transitSchedules.db.queries.ts

```diff
@@ -101,128 +106,291 @@ const scheduleTripsAttributesParser = function (attributes: any): SchedulePeriod
     };
 };
 
-const createFromTrip = async function (periodTrip: SchedulePeriodTrip, scheduleId: string, periodId: string) {
-    if (!periodTrip.id) {
-        periodTrip.id = uuidV4();
-    }
-    if (!periodTrip.schedule_id) {
-        periodTrip.schedule_id = scheduleId;
-    }
-    if (!periodTrip.schedule_period_id) {
-        periodTrip.schedule_period_id = periodId;
-    }
-    const id = (await create(knex, tripTable, scheduleTripsAttributesCleaner, periodTrip, {
-        returning: 'id'
-    })) as string;
-    return id;
+const _createTrips = async function (
+    periodTrips: SchedulePeriodTrip[],
+    options: { transaction: Knex.Transaction }
+): Promise<{ id: string }[]> {
+    const ids = (await createMultiple(knex, tripTable, scheduleTripsAttributesCleaner, periodTrips, {
+        returning: 'id',
+        transaction: options.transaction
+    })) as { id: string }[];
+    return ids;
 };
 
-const createFromPeriod = async function (periodSchedule: SchedulePeriod, schedule_id: string) {
-    if (!periodSchedule.id) {
-        periodSchedule.id = uuidV4();
-    }
-    if (!periodSchedule.schedule_id) {
-        periodSchedule.schedule_id = schedule_id;
-    }
-    const id = (await create(knex, periodTable, schedulePeriodsAttributesCleaner, periodSchedule, {
-        returning: 'id'
-    })) as string;
-    if (periodSchedule.trips) {
-        for (let i = 0; i < periodSchedule.trips.length; i++) {
-            await createFromTrip(periodSchedule.trips[i], schedule_id, id);
+const _createPeriods = async function (
+    schedulePeriods: SchedulePeriod[],
+    scheduleId: string,
+    options: { transaction: Knex.Transaction }
+): Promise<{ id: string }[]> {
+    schedulePeriods.forEach((period) => {
+        if (!period.id) {
+            period.id = uuidV4();
         }
-    }
-    return id;
+        period.schedule_id = scheduleId;
+    });
+
+    const ids = (await createMultiple(knex, periodTable, schedulePeriodsAttributesCleaner, schedulePeriods, {
+        returning: 'id',
+        transaction: options.transaction
+    })) as { id: string }[];
+    const tripPromises = schedulePeriods
+        .filter((period) => period.trips !== undefined)
+        .map((period) => {
+            period.trips.forEach((trip) => {
+                if (!trip.id) {
+                    trip.id = uuidV4();
+                }
+                trip.schedule_id = scheduleId;
+                trip.schedule_period_id = period.id;
+            });
+            return _createTrips(period.trips, options);
+        });
+    await Promise.all(tripPromises);
+    return ids;
 };
 
-// create all needed inserts from scheduleData (from scheduleByServiceId[serviceId])
-const createFromScheduleData = async function (scheduleData: ScheduleAttributes) {
+const _createFromScheduleData = async function (
```

Comment:
As usual, we have a private and public function with the same name. Please explain